### ISSUE TITLE
en-US localization review

### ISSUE THREAD

**Issue by flodolo** [mozilla-iot/gateway#2326]
# Major Issues

## Concatenations

This is not OK for localization.

```
add-thing-add-adapters-hint = No new things found. Try
add-thing-add-adapters-hint-anchor = adding some add-ons.
```

```
<p id="add-adapters-hint" class="hidden">
  <span data-l10n-id="add-thing-add-adapters-hint"></span>
  <a id="add-adapters-hint-anchor" href="/settings/addons"
     data-l10n-id="add-thing-add-adapters-hint-anchor"></a>
</p>
```		

You're concatenating two strings with an hard-coded space in-between, one of them is a link, and you're forcing the English structure to all other languages. There should be a way to avoid that, depending on the version of Fluent you're using.

This one was obvious, I'm not sure if I missed others in the file. For example, `new-web-thing-from` seems suspect, but I couldn't figure out how it's used in the code. The `Local Token Page` section is also suspicious, but I can't find any reference in the code.

## Content Exposed for Localization

```
thing-icons-thing-src = /optimized-images/thing-icons/thing.svg
```

Why is this file path exposed for localization?

## Branding

At the beginning of the file there are two messages

```
# Main Title
webthings-gateway = WebThings Gateway

# Wordmark
wordmark =
  .alt = Mozilla IoT
```

Is it correct to assume that they should not be localizable? If that's the case, it would be better to define them as terms.

```
## The following terms must be treated as brand, and kept in English.
##
## They cannot be:
## - Transliterated.
## - Translated.
##
## Declension should be avoided where possible.
##
## Reference: https://mozilla-l10n.github.io/styleguides/mozilla_general/index.html#brands-copyright-and-trademark

-webthings-gateway-brand = WebThings Gateway
-mozilla-iot-brand = Mozilla IoT

# Main Title
webthings-gateway = { -webthings-gateway-brand }

# Wordmark
wordmark =
  .alt = { -mozilla-iot-brand }
```

Even if they're not localizable, you can still use them later to make sure the product is localized consistently (only a couple of examples, there are more).

```
login-title = Login — { -webthings-gateway-brand }
signup-title = Create User — { -webthings-gateway-brand }
```

# Medium Issues

## Lack of Comments

The rules section definitely needs some comments. "and" and "or" are lowercase, why? How is "Not" used, given it's uppercase?

The *Authorize Page* section is a mistery to me. How are all those strings used together?

# Minor Issues

## Message IDs

In Firefox we enforce a kebab-case style, the file is currently a mix. Not really an issue, but it might be something you want to look into at some point.

## Typography

Currently using a straight single quote `'` instead of an apostrophe `’`. The latter is the norm in Mozilla products.

## FTL formatting

* After a section comment there should be an empty line.
* In select expressions, the square bracket should be aligned.

```
utils-seconds-ago =
  { $value ->
      [one] { $value } second ago
     *[other] { $value } seconds ago
  }
```

**Comment by flodolo** [comment]
CC @peiying2

**Comment by flodolo** [comment]
Found other concatenations while reviewing the Italian translation

```
reclaim-prompt = It looks like you've already registered that subdomain. To reclaim it
click-here = click here

local-token-use-it = Use it to talk to the gateway securely, with
local-token-bearer-type = Bearer-type Authorization
```

And one question: is **Web Thing** localizable?

**Comment by flodolo** [comment]
Per conversation on Slack (cc @pike to confirm), fluent-dom should support the same DOM overlays as Firefox.
https://github.com/projectfluent/fluent.js/wiki/DOM-Overlays

**Comment by flodolo** [comment]
In the first example, I would expect this to work

```
<p id="add-adapters-hint" class="hidden" data-l10n-id="add-thing-add-adapters">
  <span data-l10n-name="add-adapters-hint"></span>
  <a id="add-adapters-hint-anchor" href="/settings/addons"
     data-l10n-name="add-adapters-link"></a>
</p>
```

```
add-thing-add-adapters = <span data-l10n-name="add-adapters-hint">No new things found. Try</span> <a data-l10n-name="add-adapters-link">adding some add-ons.</a>
```

**Comment by mrstegeman** [comment]
Thank you for the thorough review! I'm working through these piece by piece. I'll leave comments as I go.

> # Minor Issues
> ## Message IDs
> 
> In Firefox we enforce a kebab-case style, the file is currently a mix. Not really an issue, but it might be something you want to look into at some point.

I believe the only time we don't use kebab-case is when we TitleCase some terms that come directly from our [schemas](https://iot.mozilla.org/schemas/). We'll probably leave those in place.

> ## Typography
> 
> Currently using a straight single quote `'` instead of an apostrophe `’`. The latter is the norm in Mozilla products.

Done: #2332

> ## FTL formatting
> 
> * After a section comment there should be an empty line.

Done: #2334 

> * In select expressions, the square bracket should be aligned.
> 
> 
> ```
> utils-seconds-ago =
>   { $value ->
>       [one] { $value } second ago
>      *[other] { $value } seconds ago
>   }
> ```

Done: #2333

**Comment by mrstegeman** [comment]
> # Medium Issues
> ## Lack of Comments
> 
> The rules section definitely needs some comments. "and" and "or" are lowercase, why? How is "Not" used, given it's uppercase?
> 
> The _Authorize Page_ section is a mistery to me. How are all those strings used together?

I believe these are all artifacts of our poor concatenations. Those will all be addressed in other reviews.

**Comment by mrstegeman** [comment]
> ## Content Exposed for Localization
> 
> ```
> thing-icons-thing-src = /optimized-images/thing-icons/thing.svg
> ```
> 
> Why is this file path exposed for localization?

This will be addressed by #2303.

**Comment by mrstegeman** [comment]
> ## Branding
> 
> At the beginning of the file there are two messages
> 
> ```
> # Main Title
> webthings-gateway = WebThings Gateway
> 
> # Wordmark
> wordmark =
>   .alt = Mozilla IoT
> ```
> 
> Is it correct to assume that they should not be localizable? If that's the case, it would be better to define them as terms.
> 
> ```
> ## The following terms must be treated as brand, and kept in English.
> ##
> ## They cannot be:
> ## - Transliterated.
> ## - Translated.
> ##
> ## Declension should be avoided where possible.
> ##
> ## Reference: https://mozilla-l10n.github.io/styleguides/mozilla_general/index.html#brands-copyright-and-trademark
> 
> -webthings-gateway-brand = WebThings Gateway
> -mozilla-iot-brand = Mozilla IoT
> 
> # Main Title
> webthings-gateway = { -webthings-gateway-brand }
> 
> # Wordmark
> wordmark =
>   .alt = { -mozilla-iot-brand }
> ```
> 
> Even if they're not localizable, you can still use them later to make sure the product is localized consistently (only a couple of examples, there are more).
> 
> ```
> login-title = Login — { -webthings-gateway-brand }
> signup-title = Create User — { -webthings-gateway-brand }
> ```

Done: #2335

**Comment by mrstegeman** [comment]
Concatenations part 1: #2336

**Comment by mrstegeman** [comment]
Ok, I believe everything here has been addressed, other than the one icon issue in #2303. Closing
